libqtile/layout/xmonad.py

```diff
@@ -1005,3 +1006,290 @@ def cmd_swap_main(self):
+            return
+
+        max_height = self.screen_rect.height - ((column.count - 1) * self.min_secondary_size)
+        for i in range(column.start, column.end):
+            self.relative_sizes[i] = max_height if i == focused else self.min_secondary_size
+
+    def _grow_secondary(self, amt):
+        """Grow the focused client in the secondary pane"""
+        self._resize_secondary(amt)
+
+    def _shrink_secondary(self, amt):
+        """Shrink the focused client in the secondary pane"""
+        self._resize_secondary(-amt)
+
+    def _resize_secondary(self, amt):
+        """Resize the focused secondary client
+
+        If amt is positive, the client will grow. Conversely, if it's negative,
+        the client will shrink. All other clients in the same column will get
+        grown/shrunk so to accommodate the new height.
+        """
+        focused = self.focused - 1
+        column = self._get_column(focused)
+
+        if column.count == 1:
+            return
+
+        # Resizing is accomplished by doing the following:
+        # - calculate how much each client in the column must shrink/grow
+        #   so that the focused window can grow/shrink.
+        # - iterate over all clients in the column and change their height
+        #   (grow or shrink) as long as they can still be resized (both main
+        #   and secondary windows).
+        min_height = self.min_secondary_size
+        idx = column.start
+        step = amt // (column.count - 1)
+        visited = 0
+        while amt != 0:
+            if idx != focused:
+                focused_new_height = self.relative_sizes[focused] + step
+                new_height = self.relative_sizes[idx] - step
+                if focused_new_height >= min_height and new_height >= min_height:
+                    self.relative_sizes[focused] += step
+                    self.relative_sizes[idx] -= step
+                    amt -= step
+                    visited += 1
+
+            idx += 1
+            if idx == column.end:
+                if visited == 0:
+                    break
+
+                idx = column.start
+                visited = 0
+
+        self.group.layout_all()
+
+    def _get_column(self, index):
+        """Get the column containing the provided index"""
+        left, right = self._get_columns()
+        return left if index < left.count else right
+
+    def _get_columns(self):
+        """Get all columns"""
+        Column = namedtuple("Column", "name count start end")
+        clients = len(self.clients) - 1
+        clients = [clients // 2 + clients % 2, clients // 2]
+        return [
+            Column(
+                name="left",
+                count=clients[0],
+                start=0,
+                end=clients[0],
+            ),
+            Column(
+                name="right",
+                count=clients[1],
+                start=clients[0],
+                end=clients[0] + clients[1],
+            )
+        ]
```

Comment:
Could you make this a tuple? It will always return two columns so makes it more clear.